File: pmd/java/listas/listaex01/ex3/BancoInterface.java
package listas.listaex01.ex3;

import java.util.Scanner;
import listaex01.ex2.Conta;
import listaex01.ex2.ContaInterface;

public class BancoInterface {

  private Banco banco;

  public BancoInterface(Banco banco) {
    this.banco = banco;
  }

  public void menu(Scanner scn) {
    while (true) {

      System.out.print(
          "\n0 - SAIR\n1 - ABRIR CONTA\n2 - ACESSAR CONTA\nINFORME A SUA OPÇÃO: ");
      int op = scn.nextInt();

      switch (op) {

        case 0:
          return;

        case 1:
          System.out.print("\nDigite o número e o saldo da conta: ");
          Conta conta = new Conta(scn.nextInt(), scn.nextDouble());
          banco.cadastrarConta(conta);
          break;

        case 2:
          System.out.println("\nINFORME O NÚMERO DA CONTA: ");
          int numero = scn.nextInt();
          for (Conta c : banco.getContas()) {
            if (c.getNumero() == numero) {
              ContaInterface.menu(c, scn);
              break;
            }
          }
      }
    }
  }
}

File: pmd/python/exercicios/listaex01/ex3/bancointerface.py
import sys
sys.path.insert( 1, '../ex2' )
#Os comandos abaixo significam: do arquivo x importar a classe y
from conta import Conta
from containterface import ContaInterface
from banco import Banco


class BancoInterface:

    def __init__(self, banco):
        self.banco = banco
        
    def executar(self):
        while True:
            menu =  '\n\n0 - Sair\n1 - Criar conta' + \
                    '\n2 - Selecionar conta\nDigite sua opcao: '

            op = int(input(menu))

            if op == 0:
                return
            
            elif op == 1:
                n = int( input('Informe o numero da nova conta: ') )
                s = float( input('Informe o saldo inicial da nova conta: ') )
                self.banco.addConta( Conta(n,s)  )
                print('\nConta criada com sucesso.') 
                
            elif op == 2:
                n = int( input('Informe o numero da conta: ') )
                conta = self.banco.buscarConta(n)
                if conta is None:
                    print('\nConta nao encontrada.')
                else:
                    ContaInterface.executar( conta )
                
            else:
                print('\nOpcao invalida.')
